ios/SourceModule.swift

```diff
@@ -32,15 +35,33 @@ class SourceModule: NSObject, RCTBridgeModule {
         sources[nativeId]
     }
 
+    // Finds `NativeId` based on predicate ran on `Source` instances
+    func nativeId(where predicate: (Source) -> Bool) -> NativeId? {
+        sources.first { _, value in
+            predicate(value)
+        }?.key
+    }
+
+    // Fetches cast-specific `SourceConfig` by `NativeId` if exists
+    func retrieveCastSourceConfig(_ nativeId: NativeId) -> SourceConfig? {
+        castSourceConfigs[nativeId]
+    }
+
     /**
      Creates a new `Source` instance inside the internal sources using the provided `config` and `analyticsSourceMetadata` object and an optionally initialized DRM configuration ID.
      - Parameter nativeId: ID to be associated with the `Source` instance.
      - Parameter drmNativeId: ID of the DRM config object to use.
      - Parameter config: `SourceConfig` object received from JS.
      - Parameter analyticsSourceMetadata: `SourceMetadata` object received from JS.
      */
-    @objc(initWithAnalyticsConfig:drmNativeId:config:analyticsSourceMetadata:)
-    func initWithAnalyticsConfig(_ nativeId: NativeId, drmNativeId: NativeId?, config: Any?, analyticsSourceMetadata: Any?) {
+    @objc(initWithAnalyticsConfig:drmNativeId:config:sourceRemoteControlConfig:analyticsSourceMetadata:)
+    func initWithAnalyticsConfig(
+        _ nativeId: NativeId,
+        drmNativeId: NativeId?,
+        config: Any?,
+        sourceRemoteControlConfig: Any?,
```

Comment:
Nit: I think we should also update the doc and include the new parameter. But also fine for me to remove the whole doc altogether if it is not really adding value (as this is not really public API). Same for down below

Reply:
fixed in da16514